lib/AST/ParameterPack.cpp

```diff
@@ -176,17 +176,12 @@ void TypeBase::getTypeParameterPacks(
   });
 }
 
-bool TypeBase::isParameterPack() {
-  Type t(this);
-
-  while (auto *memberTy = t->getAs<DependentMemberType>())
-    t = memberTy->getBase();
-
-  return t->isRootParameterPack();
+bool TypeBase::isParameterPack() const {
+  return getDependentMemberRoot()->isRootParameterPack();
```

Comment:
s/getDependentMemberRoot()/getRootGenericParam()/

Reply:
`getRootGenericParam()` might return a null pointer. I don’t think a client would expect this `TypeBase` method to crash if not given something rooted on a generic parameter type.